### ISSUE TITLE
SemanticsService not read by all Screen Readers on WEB.

### ISSUE THREAD

**Issue by ipainchaud** [flutter/flutter#110204]
## Steps to Reproduce

1. have a Screen Reader active on your computer
2. make the use of `SemanticsService.announce('bla')` on WEB.
3. repeat with another Screen Reader

I've tested with both JAWs (Windows) and VoiceOver (MacOS), but there are also some other screen reader which could be tested like NVDA (Windows).

**Expected results:** The String announced by `SemanticService` should be read by all Screen Readers

**Actual results:** JAWs is not reading any announcement made by `SemanticsService`.

JAWs is usually the Screen Reader which succeeds best at reading WEB content generated by Flutter, but it fails with `SemanticsService.announce`.

For an easy test, you guys can run this [code sample on dartpad.dev
](https://dartpad.dev/?id=09670c3c96195019e0a87c6031525bd9) with JAWs and VoiceOver.

There are no error log displayed at all concerning this in the console.

```
$ flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5.1 21G83 darwin-x64, locale en-US)
    • Flutter version 3.0.5 at /usr/local/Caskroom/flutter/3.0.2/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (6 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/israelpainchaud/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.5.1 21G83 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 104.0.5112.101

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by huycozy** [comment]
Hi @ipainchaud, thanks for filing the issue.
It seems the issue only happens on Windows, right? I can reproduce it on Windows using NVDA. 
For macOS with VoiceOver, the issue is not reproducible.

- Windows: ❌
- macOS: ✅

✅: No Issue ❌: Issue reproduced

<details>
<summary>Demo</summary>

| Windows (NVDA) | MacOS (VoiceOver) |
| --------------- | --------------- |
<video src="https://user-images.githubusercontent.com/104349824/186607984-c98d729d-70d7-4e41-94dd-289aa97378eb.mp4"/> | <video src="https://user-images.githubusercontent.com/104349824/186607969-1b77c290-715c-492c-a41a-1bc3576ae065.mp4"/>

</details>

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'SemanticsService Issue',
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatefulWidget {
  @override
  State<MyWidget> createState() => _MyWidgetState();
}

class _MyWidgetState extends State<MyWidget> {
  bool _value = false;
  final String checkboxLabel = 'please check and un-check me';
  
  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Text(
          'SemanticsService Issue',
          style: Theme.of(context).textTheme.headline4,
        ),
        const SizedBox(height: 80),
        const Text(
          'Here is a checkbox which will trigger some announcement when it is checked and unchecked. Those announcements can be heard when using a Screen Reader as JAWs, NVDA or VoiceOver.',
        ),
        const SizedBox(height: 40),
        Row(children: [
          Semantics(
            label: checkboxLabel,
            child: Checkbox(
              onChanged: (_) => _toggleValue(),
              value: _value,
            ),
          ),
          ExcludeSemantics(
            child: InkWell(
              canRequestFocus: false, // Avoid a second tab stop.
              onTap: () => _toggleValue(),
              child: Text(checkboxLabel),
            ),
          ),
        ],),
      ]
    );
  }
  
  void _toggleValue() {
    setState(() => _value = !_value);
    SemanticsService.announce('$checkboxLabel ${_value ? 'checked' : 'unchecked'}', TextDirection.ltr);
  }

}

```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Version 10.0.19043.1766], locale en-US)
    • Flutter version 3.0.5 at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (9 hours ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32616.157
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.69.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.44.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1766]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 103.0.5060.114
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 101.0.1210.53

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```bash
[√] Flutter (Channel master, 3.1.0-0.0.pre.2481, on Microsoft Windows [Version 10.0.19043.1889], locale en-US)
    • Flutter version 3.1.0-0.0.pre.2481 on channel master at C:\WIP\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 91fdbfa7c8 (50 minutes ago), 2022-08-25 03:07:06 -0400
    • Engine revision 9600c6c39b
    • Dart version 2.19.0 (build 2.19.0-137.0.dev)
    • DevTools version 2.16.0

Checking Android licenses is taking an unexpectedly long time...[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.2.32630.192
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.69.2)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.46.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1889]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 104.0.5112.102
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 103.0.1264.77

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by ipainchaud** [comment]
That's right! I had not tried with NVDA, but I know JAWs has the issue and VoiceOver works fine. I'm happy you were able to reproduce.

**Comment by huycozy** [comment]
@ipainchaud thanks for your confirmation. Labeling this issue for further investigation from the team.

**Comment by huycozy** [comment]
@ipainchaud thanks for your confirmation. Labeling this issue for further investigation from the team.
The issue is reproducible on the latest `stable` and `master` channels. The sample code can be found at https://github.com/flutter/flutter/issues/110204#issuecomment-1226912849.

**Comment by yjbanov** [comment]
Does the new `assertiveness` option help?

https://github.com/flutter/flutter/blob/b7c0fd773e3276c2f823cf6748ed008d427e30f8/packages/flutter/lib/src/semantics/semantics_service.dart#L36

This is probably still only on the master channel.

**Comment by ipainchaud** [comment]
I haven't been able to see the `assertiveness` parameter from Dartpad's beta channel. 
<img width="509" alt="image" src="https://user-images.githubusercontent.com/5298744/187511994-0a9e9bd6-5c44-4ec3-a1d0-dbd86c8d98d5.png">
I'll have a look soon. Sorry for the delay.

**Comment by yjbanov** [comment]
Yeah, it probably hasn't reached beta yet.
